### ISSUE TITLE
Select which files to delete in `huggingface-cli delete-cache`

### ISSUE THREAD

**Issue by sealad886** [huggingface/huggingface_hub#2219]
**Is your feature request related to a problem? Please describe.**
The download tools all either dynamically or configurably download parts of a repo, since there are many times when the repo contains more files than needed (e.g. .pt, .safetensors). Once they're downloaded, you can't delete only part of the repo from your cache, and if you accidentally (or because you didn't know what you were doing) downloaded huge volumes of files, you can't delete just a part of it. The only option right now is to delete the whole repo and then re-download only what you want in your cache. 

**Describe the solution you'd like**
Option to select files within a given cached repo, using the existing tools, that are to be deleted. 

**Describe alternatives you've considered**
Manually manipulating files in the cache has gone..let's say "poorly" in the past. I've had to scrap my entire cache more than once doing that. 

**Additional context**
n/a

**Comment by Wauplin** [comment]
This is a valid feature request indeed. At the moment we can only delete specific revisions (usually `main` for the whole repo, or outdated revisions if more than one revision). To delete files we would need to update the UI / the API. Maybe it's fine to just have it for 1 repo at a time: `huggingface-cli delete-cache <repo_id> --delete=...` or `huggingface-cli delete-cache <repo_id> --keep=...` (where delete/keep are glob patterns).

**Comment by sealad886** [comment]
Yeah, for sure I can. This would be my first time doing anything more than very superficial dev on a project like this (on GH, that is). Please be gentle :)
I can have a think through the design of it first. Hopefully I'll have some downtime from other side projects to be working on tihis. Off the cuff thoughts:
* `delete-cache` doesn't take arguments or flags in other contexts, so introducing that might be confusing.
* `delete-cache` also is going down the road of using that TUI system for multiple selection, which would be confusing / difficult to do nested.
* Perhaps a new subcommand like `prune-cache` or `trim-cache` or `trim-repo`... `repo-cache`. I like `repo-cache` because then you could extend that to more tools that would specific to managing the cache for a single repo.

Going with `repo-cache`, use case would be something like:
```bash
> huggingface-cli repo-cache delete --all repo_id     # same functionality as delete-cache but quick to delete just the one repo
> huggingface-cli repo-cache delete --include (glob) repo_id.  # use same glob format as download --include flag; mark files to __include__ for delete action
> huggingface-cli repo-cache delete --exclude [glob] repo_id. # same as --include, but opposite
> huggingface-cli repo-cache delete --file-list (list of files to remove) repo_id
```

I'll have a proper sit down at some point soon and see what's what.

Are there timelines I should be aware of that you know of?

**Comment by Wauplin** [comment]
Command has been revamped in https://github.com/huggingface/huggingface_hub/pull/3439 (revamp discussed in https://github.com/huggingface/huggingface_hub/issues/3432)
